source/grid/tria_accessor.cc

```diff
@@ -1489,19 +1489,39 @@ const unsigned int
 template <int structdim, int dim, int spacedim>
 void
 TriaAccessor<structdim, dim, spacedim>::set_bounding_object_indices(
-  const std::initializer_list<int> &object) const
+  const std::initializer_list<int> &new_indices) const
 {
-  this->objects().get_object(this->present_index) = object;
+  const ArrayView<int> bounding_object_index_ref =
+    this->objects().get_bounding_object_indices(this->present_index);
+
+  AssertDimension(bounding_object_index_ref.size(), new_indices.size());
+
+  unsigned int i = 0;
+  for (const auto &new_index : new_indices)
+    {
+      bounding_object_index_ref[i] = new_index;
```

Comment:
You could just combine this with the next line to
```suggestion
      bounding_object_index_ref[i++] = new_index;
```
:wink:

Reply:
We've generally tried to use indexing with side effects. It's hard to read this kind of code unless you're a hardcore C programmer. Most people also don't know/remember the difference between `i++` and `++i`, and so have to think harder about the correctness of the code than is necessary if the index is incremented in a separate statement.

Reply:
Yes, I know. I was not really serious about suggesting this as a change.